File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {

		int n = numberToCheck;
		//System.out.println("numberToCheck="+numberToCheck);
		String digits = Integer.toString(numberToCheck);
		int e = digits.length();
		//System.out.println("e="+e);
		int s = 0;
		for (int i = 0; i < e; i++){
			int d = Integer.parseInt(digits.substring(i,i+1));
			s += (int) Math.pow(d,e);
			//System.out.println("d^e="+Math.pow(d,e));
			//System.out.println("s="+s);
		}
		if (s == numberToCheck) return (true);
		else return (false);
	}
	//public void ArmstrongNumbers ()
}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdio.h>
#include <math.h>
bool is_armstrong_number(int candidate){
	int nd = ceil(log10(candidate));
	int s = 0;
	int ds = candidate;
	for(int i=1;i<=nd;i++){
		int d=ds%10;
		ds=ds/10;
		s+=pow(d,nd);
	}
	if (s==candidate) return true;
	return false;
}
